src/modules/authorization/reducer.ts

```diff
@@ -89,10 +116,23 @@ export function authorizationReducer(
       switch (transaction.actionType) {
         case GRANT_TOKEN_SUCCESS: {
           const { authorization } = transaction.payload
+          let authorizationExists = false
+          const data = state.data.map(stateAuthorization => {
+            if (areEqual(stateAuthorization, authorization)) {
+              authorizationExists = true
+              return authorization
+            } else {
+              return stateAuthorization
+            }
+          })
```

Comment:
This section of code seems to be doing the same as above, what do you think about replacing it with a filter and a push?